scripts/amd/setup_rocm_libs.sh

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env bash
+
+set -ex
+
+CWD=$(dirname "$0")
+
+echo "Building libdrm"
+$CWD/drm.sh
+
+echo "Collecting Necessary ROCM libraries"
+if [[ -z "${ROCM_VERSION}" ]]; then
+  ROCM_VERSION=5.4.2
+fi
+if [[ -z "${TRITON_ROCM_DIR}" ]]; then
+  TRITON_ROCM_DIR=python/triton/third_party/rocm
+fi
+  
+save_IFS="$IFS"
+IFS=. ROCM_VERSION_ARRAY=(${ROCM_VERSION})
+IFS="$save_IFS"
+if [[ ${#ROCM_VERSION_ARRAY[@]} == 2 ]]; then
+    ROCM_VERSION_MAJOR=${ROCM_VERSION_ARRAY[0]}
+    ROCM_VERSION_MINOR=${ROCM_VERSION_ARRAY[1]}
+    ROCM_VERSION_PATCH=0
+elif [[ ${#ROCM_VERSION_ARRAY[@]} == 3 ]]; then
+    ROCM_VERSION_MAJOR=${ROCM_VERSION_ARRAY[0]}
+    ROCM_VERSION_MINOR=${ROCM_VERSION_ARRAY[1]}
+    ROCM_VERSION_PATCH=${ROCM_VERSION_ARRAY[2]}
+else
+    echo "Unhandled ROCM_VERSION ${ROCM_VERSION}"
+    exit 1
+fi
+
+ROCM_INT=$(($ROCM_VERSION_MAJOR * 10000 + $ROCM_VERSION_MINOR * 100 + $ROCM_VERSION_PATCH))
+OS_NAME=`awk -F= '/^NAME/{print $2}' /etc/os-release`
+
+if [[ $ROCM_INT -eq 50402 ]]; then
+    if [[ "$OS_NAME" == *"CentOS Linux"* ]]; then
+        ROCM_PKGS=(
+            "https://repo.radeon.com/rocm/rhel8/5.4.2/main/hsa-rocr5.4.2-1.7.0.50402-104.el8.x86_64.rpm"
+            "https://repo.radeon.com/rocm/rhel8/5.4.2/main/hip-runtime-amd-5.4.22803.50402-104.el8.x86_64.rpm"
+            "https://repo.radeon.com/rocm/rhel8/5.4.2/main/hip-devel-5.4.22803.50402-104.el8.x86_64.rpm"
```

Comment:
This is pinned to a specific rocm version. We should make this a general soltion.

Reply:
This is now a general solution and will pull the libraries from any manylinux image for building the wheel https://github.com/ROCmSoftwarePlatform/triton/pull/354